Identity/AzureStack.Identity.psm1

```diff
@@ -68,6 +68,191 @@ function New-AzsAdGraphServicePrincipal {
+   .DESCRIPTION
+      Create an Azure Resource Manager Account and select given subscription
+   .EXAMPLE
+      $azureAccount = Initialize-AzureRmUserAccount -AzureEnvironment $azureEnvironment -DirectoryTenantName "azurestacktest.onmicrosoft.com"
+#>
+function Initialize-AzureRmUserAccount {
+    [CmdletBinding()]
+    Param
+    (
+        [Parameter(Mandatory = $true)]
+        [ValidateNotNull()]
+        [Microsoft.Azure.Commands.Profile.Models.PSAzureEnvironment]$AzureEnvironment,
+
+        [Parameter(Mandatory = $true)]
+        [ValidateNotNullOrEmpty()]
+        [String] $DirectoryTenantName,
+
+        # The identifier of the Administrator Subscription. If not specified, the script will attempt to use the set default subscription.
+        [Parameter()]
+        [String] $SubscriptionId = $null,
+
+        # The display name of the Administrator Subscription. If not specified, the script will attempt to use the set default subscription.
+        [Parameter()]
+        [String] $SubscriptionName = $null,
+
+        [Parameter()]
+        [PSCredential] $AutomationCredential = $null
+    )
+    # resolve the directory tenant ID from the name
+    $directoryTenantId = (New-Object uri(Invoke-RestMethod "$($AzureEnvironment.ActiveDirectoryAuthority.TrimEnd('/'))/$DirectoryTenantName/.well-known/openid-configuration").token_endpoint).AbsolutePath.Split('/')[1]
+    $params = @{
+        EnvironmentName = $AzureEnvironment.Name
+        TenantId        = $directoryTenantId
+    }
+    if ($AutomationCredential) { $params += @{ Credential = $AutomationCredential } }
+
+    # Prompts the user for interactive login flow if automation credential is not specified
+    $azureAccount = Add-AzureRmAccount @params
+
+    if ($SubscriptionName) {
+        Select-AzureRmSubscription -SubscriptionName $SubscriptionName | Out-Null
+    } elseif ($SubscriptionId) {
+        Select-AzureRmSubscription -SubscriptionId $SubscriptionId | Out-Null
+    }
+
+    return $azureAccount
+}
+
+function Get-RefreshToken {
+    [CmdletBinding()]
+    Param
+    (
+        [Parameter(Mandatory = $true)]
+        [ValidateNotNull()]
+        [Microsoft.Azure.Commands.Profile.Models.PSAzureEnvironment]$AzureEnvironment,
+
+        [Parameter(Mandatory = $true)]
+        [ValidateNotNullOrEmpty()]
+        [String] $DirectoryTenantName,
+
+        # The identifier of the Administrator Subscription. If not specified, the script will attempt to use the set default subscription.
+        [Parameter()]
+        [String] $SubscriptionId = $null,
+
+        # The display name of the Administrator Subscription. If not specified, the script will attempt to use the set default subscription.
+        [Parameter()]
+        [String] $SubscriptionName = $null,
+
+        [Parameter()]
+        [PSCredential] $AutomationCredential = $null
+    )
+
+    $azureAccount = Initialize-AzureRmUserAccount @PSBoundParameters
```

Comment:
this helper command exists too. but carefull using @PSBoundParameters like this it, it will break for example if user specifics the $AutomationCredential parameter as it doesn't exist on the cmdlet.

Reply:
Oh I see this is the other helper function, I was thinking it was Add-AzureRmAccount.